src/actions/attack.cpp

```diff
@@ -1604,6 +1604,57 @@ int under_leadership(const unit &u, const map_location& loc, const_attack_ptr we
 	return leader_effect.get_composite_value();
 }
 
+static void alignment_list(unit_ability_list& abil_list)
+{	//check if have abilities or specials alignment and remove alignment bad specified
+	const std::set<std::string> checking_alignment = {"neutral", "lawful", "chaotic", "liminal"};
+	for(unit_ability_list::iterator i = abil_list.begin(); i != abil_list.end();) {
+		if(checking_alignment.count((*i->ability_cfg)["alignment"].str()) == 0){
+			i = abil_list.erase(i);
+		} else {
+			++i;
+		}
+	}
+}
+
+static std::pair<int, bool>alignment_bonus(unit_ability_list& abil_list, const std::string& alignment, int bonus_value)
+{
+	for(const auto& i : abil_list) {
+		if((*i.ability_cfg)["alignment"].str() == alignment) {
+			return {bonus_value, true};
+		}
+	}
+	return {0, false};
+}
+
+unit_alignments::type attack_type::specials_alignment() const
```

Comment:
This is implementing a method declared in src/units/attack_type.hpp, so should be in the src/units directory.

At the moment, most of attack_type's implementation is in src/units/abilities.cpp instead of src/units/attack_type.cpp. That seems a mess, but a mess to clean up in a different PR, so for now I'd suggest putting this in src/units/abilities.cpp too.

Reply:
it's because [unit_alignment] abilitie what i want createin attack.cpp and whowill use same static functions, what i must let other in attack.cpp

Reply:
In that case the correct approach is probably to make the functions not static and declare them in the associated header.

Reply:
> In that case the correct approach is probably to make the functions not static and declare them in the associated header.

Functions moved to abilitie.cpp